VAULT: update state tracker now accounts for multiple epochs passing when advancing state enqueued withdrawn VRT

coachchucksol opened the pull request:
To address H-02, Ive added the ability to shift the state of VRTs enqueued to withdraw - this mimics how the operators state is updated over multiple epochs. 

Tests have also been added to cover this edgecase.

ebatsell commented on vault_program/src/close_update_state_tracker.rs:
```diff
@@ -42,6 +42,10 @@ pub fn process_close_vault_update_state_tracker(
+    let last_updated_epoch = vault
+        .last_full_state_update_slot()
+        .checked_div(config.epoch_length())
+        .unwrap();
```
replace unwrap

coachchucksol replied:
Thanks!

buffalu reviewed (changes requested):
can you add test for the bug this fixes?

coachchucksol commented:
The error code I've added is because we want to use checked_div. There is no way to make the epoch_length 0, so this can never happen. However, to add some clarity to it, I made a config helper function that will calculate the epoch given a slot. I have also added some - cant-be-zero checks in the integration tests for future proofing.

ebatsell commented on integration_tests/tests/vault/crank_vault_update_state_tracker.rs:
```diff
@@ -635,4 +636,378 @@ mod tests {
+        assert_eq!(depositor_token_account.amount, MINT_AMOUNT);
+    }
+
+    /// Test that the vrt withdrawal process if updating with normal crank operations
```
nit: update comment (same as above)

coachchucksol replied:
done
